<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kumi_core.xacro                | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robot">
  <ros2_control name="KumiHardware" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>
    <joint name="front_sh">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="back_sh">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="front_knee">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="back_knee">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="front_ank">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="back_ank">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
  <gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <parameters>/home/andreas/dev_ws/install/kumi_controller/share/kumi_controller/config/kumi_control_config.yaml</parameters>
    </plugin>
  </gazebo>
  <link name="body">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="2 0.3 0.3"/>
      </geometry>
      <material name="green">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="2 0.3 0.3"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.3"/>
      <inertia ixx="0.0045" ixy="0.0" ixz="0.0" iyy="0.10225" iyz="0.0" izz="0.10225"/>
    </inertial>
  </link>
  <joint name="front_sh" type="revolute">
    <origin rpy="0 0 0" xyz="1.0 0 -0.15"/>
    <parent link="body"/>
    <child link="front_leg"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="-0.7853981633974483" upper="1.5707963267948966" velocity="100"/>
  </joint>
  <transmission name="front_sh_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor1">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="front_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0.15 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
      <material name="orange">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.15 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 -0.375"/>
      <mass value="0.15"/>
      <inertia ixx="0.013625" ixy="0.0" ixz="0.0" iyy="0.013625" iyz="0.0" izz="0.002250"/>
    </inertial>
  </link>
  <joint name="front_knee" type="revolute">
    <origin rpy="0 0 0" xyz="0.15 0 -0.75"/>
    <parent link="front_leg"/>
    <child link="front_leg_d"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="100"/>
  </joint>
  <transmission name="front_knee_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor2">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="front_leg_d">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
      <material name="orange">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 -0.375"/>
      <mass value="0.15"/>
      <inertia ixx="0.013625" ixy="0.0" ixz="0.0" iyy="0.013625" iyz="0.0" izz="0.002250"/>
    </inertial>
  </link>
  <joint name="front_ank" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.9"/>
    <parent link="front_leg_d"/>
    <child link="front_foot"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="-0.7853981633974483" upper="0.7853981633974483" velocity="100"/>
  </joint>
  <transmission name="front_ank_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor3">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="front_foot">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.7 0.3 0.3"/>
      </geometry>
      <material name="white">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.7 0.3 0.3"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.2"/>
      <inertia ixx="0.003" ixy="0.0" ixz="0.0" iyy="0.009667" iyz="0.0" izz="0.009667"/>
    </inertial>
  </link>
  <gazebo reference="front_foot">
    <mu1>1.0</mu1>
    <mu2>1.0</mu2>
  </gazebo>
  <joint name="back_sh" type="revolute">
    <origin rpy="0 0 0" xyz="-1.0 0 -0.15"/>
    <parent link="body"/>
    <child link="back_leg"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="-1.5707963267948966" upper="0.7853981633974483" velocity="100"/>
  </joint>
  <transmission name="back_sh_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor4">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="back_leg">
    <visual>
      <origin rpy="0 0 0" xyz="-0.15 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
      <material name="orange">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.15 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.15 0 -0.375"/>
      <mass value="0.15"/>
      <inertia ixx="0.013625" ixy="0.0" ixz="0.0" iyy="0.013625" iyz="0.0" izz="0.002250"/>
    </inertial>
  </link>
  <joint name="back_knee" type="revolute">
    <origin rpy="0 0 0" xyz="-0.15 0 -0.75"/>
    <parent link="back_leg"/>
    <child link="back_leg_d"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="100"/>
  </joint>
  <transmission name="back_knee_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor5">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="back_leg_d">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
      <material name="orange">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.375"/>
      <geometry>
        <box size="0.3 0.3 0.75"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 -0.375"/>
      <mass value="0.15"/>
      <inertia ixx="0.013625" ixy="0.0" ixz="0.0" iyy="0.013625" iyz="0.0" izz="0.002250"/>
    </inertial>
  </link>
  <joint name="back_ank" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.9"/>
    <parent link="back_leg_d"/>
    <child link="back_foot"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="-0.7853981633974483" upper="0.7853981633974483" velocity="100"/>
  </joint>
  <transmission name="back_ank_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="motor6">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="back_foot">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.7 0.3 0.3"/>
      </geometry>
      <material name="white">
        <color rgba="0.2 1 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.7 0.3 0.3"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.2"/>
      <inertia ixx="0.003" ixy="0.0" ixz="0.0" iyy="0.009667" iyz="0.0" izz="0.009667"/>
    </inertial>
  </link>
  <gazebo reference="back_foot">
    <mu1>1.0</mu1>
    <mu2>1.0</mu2>
  </gazebo>
</robot>
